### PR TITLE
feat(serverless): send bottlecap failover metric

### PR DESCRIPTION
**Pull request by duncanista**
### What does this PR do?

Sends a non-billable metric specifying the failover reason on the Datadog Extension.

### Motivation

Get business insights on failover reason for Bottlecap.
[datadog-lambda-extension#292](https://github.com/DataDog/datadog-lambda-extension/pull/292)
Also [5004](https://datadoghq.atlassian.net/browse/SVLS-5004)

### Possible Drawbacks / Trade-offs

- We check every extension string to see if it is what we are looking for, it should be fast tho.

### Describe how to test/QA your changes
- Unit tests
- Manual failover test

### CODE REVIEW

File: pkg/serverless/logs/logs_collector.go [hunk 1]
```diff
@@ -279,6 +283,22 @@ func (lc *LambdaLogsCollector) processMessage(
 				outOfMemoryRequestId = lc.lastRequestID
 			}
 		}
+
+		// Bottlecap Failover
+		if message.logType == logTypeExtension {
+			var r map[string]interface{}
+			record := message.stringRecord
+			err := json.Unmarshal([]byte(record), &r)
```
Comment (duncanista) [inline pkg/serverless/logs/logs_collector.go:291]: Not fan of the idea of trying to unmarshal every extension log just to find the reason, is there a faster way of doing this?

Comment (astuyve) [reply pkg/serverless/logs/logs_collector.go:291]: ah yeah that's a bummer, wondering if we can grep for the env var and parse if we get a hit?

Comment (astuyve) [reply pkg/serverless/logs/logs_collector.go]: nice fix, looks good!

File: pkg/serverless/logs/logs_collector.go [hunk 2]
```diff
@@ -261,16 +265,17 @@ func (lc *LambdaLogsCollector) processMessage(
 		message.stringRecord = createStringRecordForReportLog(lc.invocationStartTime, lc.invocationEndTime, message)
 	}
 
+	proactiveInit := false
```
Comment (astuyve) [inline pkg/serverless/logs/logs_collector.go:268]: is there a reason we moved this out of the block which checks for enhanced metrics?

Comment (duncanista) [reply pkg/serverless/logs/logs_collector.go:268]: Yeah, I want to send failover metric no matter if enhanced metrics is disabled or not, what do you think?

Comment (astuyve) [reply pkg/serverless/logs/logs_collector.go:268]: Right but why did you move out the cold start and proactive init tag block?

Comment (duncanista) [reply pkg/serverless/logs/logs_collector.go:268]: Tags are initialized at the same time as cold start is being added, do you want me to update that?

Comment (astuyve) [reply pkg/serverless/logs/logs_collector.go:268]: I think I goofed this up when I first wrote it, we probably want this tag on all logs even if the user isn't using enhanced metrics.

**Review by astuyve — APPROVED** [review]

**Pull request merged**